src/BuildScriptGenerator/DotNetCore/DotnetCoreLanguageDetector.cs

```diff
@@ -67,14 +67,9 @@ public LanguageDetectorResult Detect(RepositoryContext context)
 
         private string GetVersion(RepositoryContext context, string targetFramework)
         {
-            if (context.DotNetCoreRuntimeVersion != null)
+            if (context.ResolvedDotNetCoreRuntimeVersion != null)
```

Comment:
I am changing the name to `Resolved..` here because the final version to be considered is based on the following:
- version provided through `options` (i.e command line, env variables and build.env)
- if not version was provided through options, get the detected version

So all the components should thereby work on the 'resolved' version and not on the version directly from `options`.